whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestBasicAuthTestIntegration.java

```diff
@@ -0,0 +1,273 @@
+                new Attribute("tech-c", "TP1-TEST", null, "person", Link.create("http://rest-test.db.ripe.net/test/person/TP1-TEST"), null),
+                new Attribute("auth", "MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/", "test", null, null, null),
+                new Attribute("mnt-by", "OWNER-MNT", null, "mntner", Link.create("http://rest-test.db.ripe.net/test/mntner/OWNER-MNT"), null),
+                new Attribute("source", "TEST")));
+    }
+
+    @Test
+    public void create_succeeds_with_basic_auth_no_password_query() {
+        final WhoisResources whoisResources = SecureRestTest.target(getSecurePort(), "whois/test/person")
+                .request()
+                .header("Authorization", getBasicAuthenticationHeader("test", "test"))
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), WhoisResources.class);
+
+        assertThat(whoisResources.getLink().getHref(), is(String.format("https://localhost:%s/test/person",getSecurePort())));
+        assertThat(whoisResources.getErrorMessages(), is(empty()));
+        final WhoisObject object = whoisResources.getWhoisObjects().get(0);
+
+        assertPersonObject(whoisResources, object);
+    }
+
+    @Test
+    public void create_failed_with_basic_auth_incorrect_www_authenticate_cookie() {
+        final Response response = SecureRestTest.target(getSecurePort(), "whois/test/person")
+                .request()
+                .header("Authorization", getBasicAuthenticationHeader("test", "incorrect"))
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), Response.class);
+
+        assertThat(response.getHeaderString(HttpHeaders.WWW_AUTHENTICATE), is(BASIC_CHARSET_ISO_8859_1_LATIN_1));
+    }
+
+    @Test
+    public void create_failed_with_no_basic_auth_incorrect_password_no_www_authenticate_cookie() {
+        final Response response = SecureRestTest.target(getSecurePort(), "whois/test/person?password=incorrect")
+                .request()
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), Response.class);
+
+        assertThat(response.getHeaderString(HttpHeaders.WWW_AUTHENTICATE), is(nullValue()));
+    }
+
+    @Test
+    public void create_failed_with_basic_auth_no_https() {
+
+        final Response response =  RestTest.target(getPort(), "whois/test/person")
+                    .request()
+                    .header("Authorization", getBasicAuthenticationHeader("test", "test"))
+                    .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), Response.class);
+
+        assertThat(response.getStatus(), is(HttpStatus.UPGRADE_REQUIRED_426));
+    }
+
+    @Test
+    public void create_failed_unauthorized_WWW_authenticate_header() {
+        final Response response = SecureRestTest.target(getSecurePort(), "whois/test/person?password=incorrect")
```

Comment:
Can we use assertThrows here as well?

Reply:
then change the name.
create_failed -> create_succeed ?

Reply:
Ah thanks for pointing out its a missed scenario. I will change it